avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1058,12 +1058,42 @@ private Object toConnectData(Schema schema, Object value) {
             arrayVal =
             recordValue.get(ANYTHING_SCHEMA.getField(ANYTHING_SCHEMA_ARRAY_FIELD).pos());
         if (arrayVal != null) {
-          return toConnectData(SchemaBuilder.array(null).build(), arrayVal);
+          // We cannot reuse the logic like we do in other cases because it is not possible to
+          // construct an array schema with a null item schema, but the items have no schema.
+          if (!(arrayVal instanceof Collection)) {
+            throw new DataException(
+                "Expected a Collection for schemaless array field but found a "
+                + arrayVal.getClass().getName()
+            );
+          }
+          Collection<Object> original = (Collection<Object>) arrayVal;
+          List<Object> result = new ArrayList<>(original.size());
+          for (Object elem : original) {
+            result.add(toConnectData((Schema) null, elem));
+          }
+          return result;
         }
 
         Object mapVal = recordValue.get(ANYTHING_SCHEMA.getField(ANYTHING_SCHEMA_MAP_FIELD).pos());
         if (mapVal != null) {
-          return toConnectData(SchemaBuilder.map(null, null).build(), mapVal);
+          // We cannot reuse the logic like we do in other cases because it is not possible to
+          // construct a map schema with a null item schema, but the items have no schema.
+          if (!(mapVal instanceof List)) {
```

Comment:
Sorry, I was referring to this one basically. Didn't want to leave comments in both places because I wasn't sure which option you'd prefer (if any).